Sources/Kafka/KafkaConsumer.swift

```diff
@@ -331,26 +352,89 @@ public final class KafkaConsumer: Sendable, Service {
             try self.subscribe(topics: topics)
         }
 
+        try await withThrowingTaskGroup(of: Void.self) { group in
+            group.addTask {
+                try await self.eventRunLoop()
+            }
+
+            group.addTask {
+                try await self.messageRunLoop()
+            }
+
+            // Throw when one of the two child task throws
+            try await group.next()
+            try await group.next()
+        }
+    }
+
+    /// Run loop polling Kafka for new events.
+    private func eventRunLoop() async throws {
+        while !Task.isCancelled {
+            let nextAction = self.stateMachine.withLockedValue { $0.nextEventPollLoopAction() }
+            switch nextAction {
+            case .pollForEvents(let client):
+                // Event poll to serve any events queued inside of `librdkafka`.
+                _ = client.eventPoll()
+                try await Task.sleep(for: self.configuration.pollInterval)
```

Comment:
That is obviously out of scope regarding sleeping here and likely should be done separately. 
What do you think about symmetry with consumer polls thus sleeping only in case we out of events otherwise continue to poll events?

Reply:
Yes, let's do this in a follow-up PR! This probably relates to the #128 issue